crates/macros/src/implement_macro.rs

```diff
@@ -142,7 +142,7 @@ impl ImplementMacro {
                         .resolve_type_def(namespace, name)
                         .is_public_composable()
                     {
-                        self.extend = Some((namespace, name));
+                        self.extend.replace((namespace, name));
```

Comment:
@kennykerr This one isn't complete yet, I intended to use `replace()` to check that the original value of `self.extend` is `None`:

```suggestion
                        assert!(self.extend.replace((namespace, name)).is_none());
```

Unfortunately `.expect_none("msg")` is still unstable.

Reply:
Yes, I wasn't yet sure whether I wanted to assert.

Reply:
Then it's probably fine to leave the `=`, that might be more representative.